File: backend/app/schemas/bar.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class BarRead(BaseModel):
    id: int
    osm_id: int
    osm_type: str
    name: str
    amenity: str
    latitude: float
    longitude: float
    address: str | None
    postcode: str | None
    city: str
    is_closed: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/bar.ts
export interface Bar {
  id: number;
  osm_id: number;
  osm_type: string;
  name: string;
  amenity: string;
  latitude: number;
  longitude: number;
  address: string | null;
  postcode: string | null;
  city: string;
  is_closed: boolean;
  created_at: string;
  updated_at: string;
}
